main.go

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright 2022 The Numaproj Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"cloud.google.com/go/pubsub"
+	"context"
+	"fmt"
+	"github.com/numaproj/numaflow-go/pkg/sourcer"
+	"log"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/numaproj-contrib/gcp-pub-sub-source-go/pkg/pubsubsource"
+)
+
+// getSubscription checks if the specified topic and subscription exist.
+// It returns an error if either the topic or the subscription doesn't exist or the subscription object
+func getSubscription(ctx context.Context, client *pubsub.Client, topicID, subID string) (*pubsub.Subscription, error) {
+	topic := client.Topic(topicID)
+	exists, err := topic.Exists(ctx)
+	if err != nil {
+		return nil, err
+	}
+	if !exists {
+		return nil, fmt.Errorf("topic does not exist: %s", topicID)
+	}
+	sub := client.Subscription(subID)
+	exists, err = sub.Exists(ctx)
+	if err != nil {
+		return nil, err
+	}
+	if !exists {
+		return nil, fmt.Errorf("subscription does not exist: %s", subID)
+	}
+	return sub, nil
+}
+
+func main() {
+	subscriptionId := os.Getenv("SUBSCRIPTION_ID")
+	topicId := os.Getenv("TOPIC_ID")
+	projectId := os.Getenv("PROJECT_ID")
+	maxOutStandingMessages, err := strconv.Atoi(os.Getenv("MAX_OUT_STANDING_MESSAGES"))
```

Comment:
I don't think we should expose `MAX_OUT_STANDING_MESSAGES` to users because 1. it's adding a learning curve to the user. I expect that most of the user doesn't care about it so let's not expose at least for now util we have a valid use case and 2. now that we are using this value to configure our message channel size and use the length of the channel to determine `Pending`, we should set it into a high value, one can argue we should set it as infinity. Because we expect high `Pending` to trigger auto-scaling.

For this PR, I am okay with setting it as 1000 or higher. We can figure out a better number when we load-test our source.